### PR TITLE
feature: Fixes #324 and B-193923794- Added Android Snippets for "Measure_ad_revenue" Page

### LINKED ISSUE #324: Add Snippets for measure ad revenue
Add in snippets for measure ad revenue for various mediation networks.

### PR DESCRIPTION
**Pull request by grantkemp**
Fixes #324 
Including snippets for AppLovin, Mopub and Ironsource. AppLovin code is the only new code added over the existing code shown in the Measure Ad Revenue Page in Helpcenter. https://firebase.google.com/docs/analytics/measure-ad-revenue.

### CODE REVIEW

File: analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt [hunk 1]
```diff
@@ -193,4 +193,61 @@ class MainActivity : AppCompatActivity() {
...
+    }
+    // [END ad_Impression_applovin]
+
+    // [START ad_Impression_ironsource]
+    @Override
+    public void onImpression(@NonNull final String adUnitId, @Nullable final ImpressionData impressionData) {
+
+        if (impressionData != null) {
+            // Feed impression data into internal tools or send to third-party analytics
+            FirebaseAnalytics firebaseAnalytics = FirebaseAnalytics.getInstance(context);
```
Comment (thatfiredev) [inline analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt:238]: Kotlin syntax (and KTX) :
```suggestion
            val firebaseAnalytics = Firebase.analytics(context);
```

File: analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt [hunk 2]
```diff
@@ -193,4 +193,61 @@ class MainActivity : AppCompatActivity() {
...
+                param(FirebaseAnalytics.Param.VALUE, impressionData.getRevenue())
+            }
+        }
+    }
+    // [END ad_Impression_moPub]
+
+    // [START ad_Impression_applovin]
+    override fun onAdRevenuePaid(impressionData: MaxAd) {
+        if (impressionData != null) {
+           FirebaseAnalytics firebaseAnalytics = FirebaseAnalytics.getInstance(context);
```
Comment (thatfiredev) [inline analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt:218]: Same as above

```suggestion
           val firebaseAnalytics = Firebase.analytics(context);
```

**Review by thatfiredev — COMMENTED** [review]
@grantkemp I see from your profile that you work with Swift mostly, but your Java & Kotlin are not that bad :)

I just left a few suggestions.

**Comment by grantkemp** [comment]
Thanks for the valuable feedback from @thatfiredev and @marinacoelho

File: analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt [hunk 3]
```diff
@@ -160,37 +136,82 @@ class MainActivity : AppCompatActivity() {
...
+            }
         // Promotion displayed
         analytics.logEvent(FirebaseAnalytics.Event.VIEW_PROMOTION, promoParams)
-
         // Promotion selected
         analytics.logEvent(FirebaseAnalytics.Event.SELECT_PROMOTION, promoParams)
         // [END apply_promo]
     }
+    // [START ad_impression_moPub]
+    override fun onImpression(adUnitId: String, impressionData: ImpressionData?) {
```
Comment (morganchen12) [inline analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt:168]: Based on the [superclass](https://github.com/firebase/snippets-android/blob/master/analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt#L12), `override` is not applicable (here and throughout).

**Review by morganchen12 — COMMENTED** [review]
The CI errors mostly revolve around misuse of `override`:

```
e: /home/runner/work/snippets-android/snippets-android/analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt: (168, 5): 'onImpression' overrides nothing
e: /home/runner/work/snippets-android/snippets-android/analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt: (168, 65): Unresolved reference: ImpressionData
e: /home/runner/work/snippets-android/snippets-android/analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt: (170, 67): Unresolved reference: context
e: /home/runner/work/snippets-android/snippets-android/analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt: (184, 5): 'onAdRevenuePaid' overrides nothing
e: /home/runner/work/snippets-android/snippets-android/analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt: (184, 50): Unresolved reference: MaxAd
e: /home/runner/work/snippets-android/snippets-android/analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt: (186, 67): Unresolved reference: context
e: /home/runner/work/snippets-android/snippets-android/analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt: (199, 5): 'onImpressionSuccess' overrides nothing
e: /home/runner/work/snippets-android/snippets-android/analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt: (199, 54): Unresolved reference: ImpressionData
e: /home/runner/work/snippets-android/snippets-android/analytics/app/src/main/java/com/google/firebase/example/analytics/kotlin/MainActivity.kt: (204, 63): Unresolved reference: context
```

The unresolved references should be coming from the Analytics import, not sure why Kotlin is flagging those.

**Comment by grantkemp** [comment]
Closing this off as we are revising the approach for ad_impression snippets.

**Pull request closed without merging**